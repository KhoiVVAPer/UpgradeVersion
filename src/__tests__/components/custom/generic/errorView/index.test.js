import React from 'react';
import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import ErrorView from '../../../../../components/custom/generic/errorView';

const setUp = (props) => {
  const component = shallow(<ErrorView {...props} />);
  return component;
};

describe('components > custom > elements > ErrorView', () => {
  let component;
  let props = {};
  beforeEach(() => {
    component = setUp(props);
    return component;
  });

  it('Check snapshot', () => {
    const tree = renderer.create(component).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('Renders component', () => {
    const input = component.find('ScrollView').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Check if component has refresh callback', () => {
    const input = component.find('ScrollView').prop('refreshControl');
    expect(input).toBeDefined();
  });

  it('Renders error image', () => {
    const input = component.find('FontIcon').length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders error title component', () => {
    const input = component.find({ dcaTest: 'title' }).length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders actual error component', () => {
    const input = component.find({ dcaTest: 'error' }).length;
    const output = 1;
    expect(input).toEqual(output);
  });

  it('Renders Settings button component when settings prop is true', () => {
    props = {
      settings: true
    };
    component = setUp(props);
    const input = component.find('Button').length;
    const output = 1;
    expect(input).toEqual(output);
  });
});
