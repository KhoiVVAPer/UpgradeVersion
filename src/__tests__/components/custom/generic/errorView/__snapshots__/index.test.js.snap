// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components > custom > elements > ErrorView Check snapshot 1`] = `
<react-native-mock>
  <react-native-mock>
    <react-native-mock>
      î “
    </react-native-mock>
  </react-native-mock>
  <react-native-mock>
    ERROR 404
  </react-native-mock>
  <react-native-mock>
    Sorry, this content is at the moment not availble because you are offline.
Please try again later
  </react-native-mock>
</react-native-mock>
`;
